core/src/main/java/org/elasticsearch/index/IndexModule.java

```diff
@@ -330,7 +330,8 @@ public IndexService newIndexService(
             IndicesQueryCache indicesQueryCache,
             MapperRegistry mapperRegistry,
             IndicesFieldDataCache indicesFieldDataCache,
-            NamedWriteableRegistry namedWriteableRegistry)
+            NamedWriteableRegistry namedWriteableRegistry,
+            Consumer<ShardId> globalCheckpointSyncer)
```

Comment:
remove this and remove all changes to IndicesService.createIndex etc.

Reply:
I pushed 26e4c7689b9e850df7affcd7b0881c31de1e6029.